test/multiverse/suites/aws_sdk_lambda/aws_sdk_lambda_instrumentation_test.rb

```diff
@@ -0,0 +1,118 @@
+
+  def test_client_call_raises_an_exception
+    in_transaction do |txn|
+      client = Aws::Lambda::Client.new(region: REGION)
+      client.config.account_id = AWS_ACCOUNT_ID
+      def client.process_response(*_args); raise 'kaboom'; end
+
+      assert_raises(RuntimeError) { client.invoke(function_name: 'Invoke-Me-And-Explode') }
+      noticed_error = lambda_segment(txn).noticed_error
+
+      assert_equal 'kaboom', noticed_error.message
+      assert_equal 'RuntimeError', noticed_error.exception_class_name
+    end
+  end
+
+  def test_client_response_indicates_an_unhandled_function_error
+    function_error = 'Unhandled'
+    message = 'oh no'
+    type = 'Function<RuntimeError>'
+    backtrace = ["/var/task/lambda_function.rb:4:in `lambda_handler'",
+      "/var/runtime/gems/aws_lambda_ric-3.0.0/lib/aws_lambda_ric/lambda_handler.rb:28:in `call_handler'",
+      "/var/runtime/gems/aws_lambda_ric-3.0.0/lib/aws_lambda_ric.rb:88:in `run_user_code'",
+      "/var/runtime/gems/aws_lambda_ric-3.0.0/lib/aws_lambda_ric.rb:66:in `start_runtime_loop'",
+      "/var/runtime/gems/aws_lambda_ric-3.0.0/lib/aws_lambda_ric.rb:49:in `run'",
+      "/var/runtime/gems/aws_lambda_ric-3.0.0/lib/aws_lambda_ric.rb:221:in `bootstrap_handler'",
+      "/var/runtime/gems/aws_lambda_ric-3.0.0/lib/aws_lambda_ric.rb:203:in `start'",
+      "/var/runtime/index.rb:4:in `<main>'"]
+    payload = StringIO.new(JSON.generate({'errorMessage' => message, 'errorType' => type, 'stackTrace' => backtrace}))
+    response = {status_code: 200, function_error: function_error, payload: payload}
+
+    in_transaction do |txn|
+      function_name = 'Invoke-At-Your-Own-Risk'
+      client = Aws::Lambda::Client.new(region: REGION, stub_responses: {invoke: response})
+      client.config.account_id = AWS_ACCOUNT_ID
+      client.invoke(function_name: function_name)
+      noticed_error = lambda_segment(txn).noticed_error
+
+      assert_equal "[#{function_error}] #{type} - #{message}", noticed_error.message
+      assert_equal backtrace, noticed_error.stack_trace
+    end
+  end
+
+  private
+
+  def perform_invocation(method, response = {}, extra_args = {})
+    function_name = 'Half-Life'
+    al = extra_args.delete(:alias)
+    function_name = "#{function_name}:#{al}" if al
+
+    with_config('cloud.aws.account_id': AWS_ACCOUNT_ID) do
+      in_transaction do |txn|
+        client = Aws::Lambda::Client.new(region: REGION, stub_responses: {method => response})
+        client.config.account_id = AWS_ACCOUNT_ID
+
+        client.send(method, {function_name: function_name}.merge(extra_args))
+
+        segment = lambda_segment(txn)
+
+        assert_equal("External/Lambda/#{method}/#{function_name}", segment.name)
+        assert_equal("lambda.#{REGION}.amazonaws.com", segment.host)
+        assert_equal(200, segment.http_status_code) unless method == :invoke_async
+        assert_equal('aws_sdk_lambda', segment.library) # rubocop:disable Minitest/EmptyLineBeforeAssertionMethods
```

Comment:
Is the `rubocop:disable` needed here because of the `unless` on the previous line?

Reply:
Yeah, precisely. I disagreed with the need for the blank line in this context.